service/src/test/java/com/bankbazaar/service/service/TripUserMapServiceTest.java

```diff
@@ -0,0 +1,127 @@
+        userDto.setUserName("name");
+        userDto.setPassword("password");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        userDto.setDob(date);
+        userDto.setEmail("name@mail.com");
+        userDto.setPhone("1234567891");
+        UserDto user = userService.addUser(userDto);
+
+        TripDto tripDto =new TripDto();
+        tripDto.setTripName("name");
+        tripDto.setTripDescription("trip");
+        tripDto.setDestination("trip");
+        Date startDate = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripDto.setStartDate(startDate);
+        Date endDate = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripDto.setEndDate(endDate);
+        tripService.addTrip(tripDto, user.getUserId());
+    }
+
+    @Test
+    void addMember() throws ParseException {
+        UserDto userDto =new UserDto();
+        userDto.setUserName("name");
+        userDto.setPassword("password");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        userDto.setDob(date);
+        userDto.setEmail("name2@mail.com");
+        userDto.setPhone("1234567891");
+        UserDto user = userService.addUser(userDto);
+
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        service.addMember(tripUserMap.get(0).getTripId(),user.getUserId());
+        assertEquals(tripUserMapRepository.count(),2);
+
+    }
+
+    @Test
+    void deleteMember() throws ParseException {
+        UserDto userDto =new UserDto();
+        userDto.setUserName("name");
+        userDto.setPassword("password");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        userDto.setDob(date);
+        userDto.setEmail("name2@mail.com");
+        userDto.setPhone("1234567891");
+        UserDto user = userService.addUser(userDto);
+
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        service.addMember(tripUserMap.get(0).getTripId(),user.getUserId());
+        Boolean response = service.deleteMember(tripUserMap.get(0).getTripId(),user.getUserId());
+        assertTrue(response);
+    }
+
+    @Test
+    void updateMember() throws ParseException {
+        UserDto userDto =new UserDto();
+        userDto.setUserName("name");
+        userDto.setPassword("password");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        userDto.setDob(date);
+        userDto.setEmail("name2@mail.com");
+        userDto.setPhone("1234567891");
+        UserDto user = userService.addUser(userDto);
+
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        service.addMember(tripUserMap.get(0).getTripId(),user.getUserId());
+        TripUserMapDto tripUserMapDto = service.updateMember(tripUserMap.get(0).getTripId(),user.getUserId(),"ADMIN");
+        assertEquals(UserRole.ADMIN.toString(), tripUserMapDto.getUserRole());
```

Comment:
Need a test for delete member when the admin leaves and there are no other admins so a new admin needs to be created